tests/unit/test_main.py

```diff
@@ -0,0 +1,89 @@
+from fastapi.testclient import TestClient
+from unittest.mock import patch
+from main import app
+from pagination_model import PaginationParams, PaginatedResult
+import os
+
+# Create a test client for the FastAPI app
+client = TestClient(app)
+
+
+def mock_search_issues(params):
+    # Mocked data to simulate PaginatedResult
+    return PaginatedResult(
+        params=PaginationParams(offset=params.offset, limit=params.limit),
+        total_results_available=2,
+        data=[
+            {
+                "dataset": "ancient-woodland",
+                "resource": "1d5336e3a650cb037328bf9c2911309328cdc743935842d7b6792484ccac7eca",
+                "line-number": 549,
+                "entry-number": 548,
+                "field": "geometry",
+                "issue-type": "invalid geometry - fixed",
+                "value": "Too few points in geometry component[0.259555 50.936166]",
+                "message": "None",
+            },
+            {
+                "dataset": "ancient-woodland",
+                "resource": "1d5336e3a650cb037328bf9c2911309328cdc743935842d7b6792484ccac7eca",
+                "line-number": 934,
+                "entry-number": 933,
+                "field": "geometry",
+                "issue-type": "invalid geometry - fixed",
+                "value": "Ring Self-intersection[0.124675 51.139896]",
+                "message": "None",
+            },
+        ],
+    )
+
+
+def test_bucket_and_file_existence(s3_client, s3_bucket):
```

Comment:
Why do we need to test this?  That's not a feature of our code right?

Reply:
No i just added some additional tests initially to make sure the bucket I am creating exists. You are right, it shouldn't be in unit tests. I ll remove them now